python/tests/unit/arcticdb/version_store/test_parallel.py

```diff
@@ -683,128 +904,305 @@ def test_parallel_append_dynamic_schema_missing_column(lmdb_version_store_tiny_s
     assert_frame_equal(expected, received)
 
 
+@pytest.mark.parametrize("delete_staged_data_on_failure", [True, False])
+def test_parallel_write_dynamic_schema_named_index(
+    lmdb_version_store_tiny_segment_dynamic, delete_staged_data_on_failure
+):
+    lib = lmdb_version_store_tiny_segment_dynamic
+    sym = "test_parallel_append_dynamic_schema_named_index"
+    df_0 = pd.DataFrame(
+        {"col_0": [0], "col_1": [0.5]}, index=pd.date_range("2024-01-01", periods=1)
+    )
+    df_0.index.name = "date"
+    df_1 = pd.DataFrame({"col_0": [1]}, index=pd.date_range("2024-01-02", periods=1))
+    lib.write(sym, df_0, parallel=True)
+    lib.write(sym, df_1, parallel=True)
+
+    with pytest.raises(SchemaException) as exception_info:
+        lib.compact_incomplete(
+            sym,
+            False,
+            False,
+            delete_staged_data_on_failure=delete_staged_data_on_failure,
+        )
+
+    assert "date" in str(exception_info.value)
+    expected_key_count = 0 if delete_staged_data_on_failure else 2
+    assert len(get_append_keys(lib, sym)) == expected_key_count
+
+
+@pytest.mark.parametrize("delete_staged_data_on_failure", [True, False])
+def test_parallel_append_dynamic_schema_named_index(
+    lmdb_version_store_tiny_segment_dynamic, delete_staged_data_on_failure
+):
+    lib = lmdb_version_store_tiny_segment_dynamic
+    sym = "test_parallel_append_dynamic_schema_named_index"
+    df_0 = pd.DataFrame(
+        {"col_0": [0], "col_1": [0.5]}, index=pd.date_range("2024-01-01", periods=1)
+    )
+    df_0.index.name = "date"
+    df_1 = pd.DataFrame({"col_0": [1]}, index=pd.date_range("2024-01-02", periods=1))
+    lib.write(sym, df_0)
+    lib.append(sym, df_1, incomplete=True)
```

Comment:
I think `append(...incomplete=True)` is the same as `write(...incomplete=True)`. Let's keep the test. It's questionable design decision that we allow it.